### ISSUE TITLE
Can't read `filename` field from tracepoint:sched:sched_process_exec

### ISSUE THREAD

**Issue by aktau** [iovisor/bpftrace#999]
The definition of the tracepoint:

```
# cat /sys/kernel/debug/tracing/events/sched/sched_process_exec/format
name: sched_process_exec
ID: 307
format:
	field:unsigned short common_type;	offset:0;	size:2;	signed:0;
	field:unsigned char common_flags;	offset:2;	size:1;	signed:0;
	field:unsigned char common_preempt_count;	offset:3;	size:1;	signed:0;
	field:int common_pid;	offset:4;	size:4;	signed:1;

	field:__data_loc char[] filename;	offset:8;	size:4;	signed:1;
	field:pid_t pid;	offset:12;	size:4;	signed:1;
	field:pid_t old_pid;	offset:16;	size:4;	signed:1;

print fmt: "filename=%s pid=%d old_pid=%d", __get_str(filename), REC->pid, REC->old_pid
```

Fails to print `filename`:

```
# bpftrace -e 'tracepoint:sched:sched_process_exec { printf("filename=%s rpid=%d pid=%d old_pid=%d\n", args->filename, pid, args->pid, args->old_pid); }'
Struct/union of type '_tracepoint_sched_sched_process_exec' does not contain a field named 'filename'
```

Apparently it's not a problem to read `old_pid` and `pid` though. If I replace `args->filename` with "N/A":

```
# # bpftrace -e 'tracepoint:sched:sched_process_exec { printf("filename=%s rpid=%d pid=%d old_pid=%d\n", "N/A", pid, args->pid, args->old_pid);  }'
Attaching 1 probe...
filename=N/A rpid=120995 pid=120995 old_pid=120995
filename=N/A rpid=120996 pid=120996 old_pid=120996
filename=N/A rpid=120997 pid=120997 old_pid=120997
```

Strangely enough. A similar field can be access on other tracepoints, like `syscalls:sys_enter_execve`:

```
# cat /sys/kernel/debug/tracing/events/syscalls/sys_enter_execve/format
name: sys_enter_execve
ID: 677
format:
	field:unsigned short common_type;	offset:0;	size:2;	signed:0;
	field:unsigned char common_flags;	offset:2;	size:1;	signed:0;
	field:unsigned char common_preempt_count;	offset:3;	size:1;	signed:0;
	field:int common_pid;	offset:4;	size:4;	signed:1;

	field:int __syscall_nr;	offset:8;	size:4;	signed:1;
	field:const char * filename;	offset:16;	size:8;	signed:0;
	field:const char *const * argv;	offset:24;	size:8;	signed:0;
	field:const char *const * envp;	offset:32;	size:8;	signed:0;

print fmt: "filename: 0x%08lx, argv: 0x%08lx, envp: 0x%08lx", ((unsigned long)(REC->filename)), ((unsigned long)(REC->argv)), ((unsigned long)(REC->envp))
```

No problem:

```
# bpftrace -e 'tracepoint:syscalls:sys_enter_execve { printf("filename=%s pid=%d\n", str(args->filename), pid); }'
Attaching 1 probe...
filename=/usr/bin/git pid=122842
filename=/usr/bin/git pid=122843
filename=/usr/bin/git pid=122844
```

The only difference I can see is in the type of the tracepoint field:

```
	field:__data_loc char[] filename;	offset:8;	size:4;	signed:1; // <---- sched:sched_process_exec
	field:const char * filename;	offset:16;	size:8;	signed:0; // <---syscalls:sys_enter_execve
```

At the least it would be better if bpftrace(1) could state that it doesn't know how to read/print/access this field, rather than saying that it doesn't exist.

**Comment by aktau** [comment]
...and only after typing this up do I realize this looks like a dup of issue #385.

**Comment by mmisono** [comment]
The current workaround is manually casting the args.

```
#include <linux/types.h>

struct data {
  long pad;
  int filename;
  pid_t pid;
  pid_t old_pid;
}

tracepoint:sched:sched_process_exec {
  $data = (struct data*)args;
  printf("filename=%s\n", str((uint64)args + (uint64)($data->filename & 0xFFFF)));
}
```

**Comment by aktau** [comment]
Awesome, thanks! Maybe this should be mentioned more prominently (google searchable?) as a current feature gap.

**Comment by mmisono** [comment]
I found that `str()` accepts optional length, so  the following is better.

```
#include <linux/types.h>

struct data {
  long pad;
  int filename;
  pid_t pid;
  pid_t old_pid;
}

tracepoint:sched:sched_process_exec {
  $data = (struct data*)args;
  $offset = (uint64)($data->filename & 0xFFFF);
  $length = (uint64)($data->filename >> 16);
  $base = (uint64)args;
  printf("filename=%s\n", str($base + $offset, $length));
}
```

**Comment by viktormalik** [comment]
Fixed by #1542.
